### ISSUE TITLE
Logs show ValueError for empty static files

### ISSUE THREAD

**Issue by juhoinkinen** [benoitc/gunicorn#2692]
GET requests for empty static files log `ValueError: count must be a positive integer (got 0)`.

Using this app:
```python
from flask import Flask, send_from_directory

app = Flask(__name__, static_url_path='')

@app.route('/static/<path:path>')
def send_static(path):
    return send_from_directory('static', path)

if __name__ == "__main__":
    app.run()
```
And these `static` dir contents:
```
static/
├── [          0]  empty-file
└── [         14]  hello-file
```

Starting with `gunicorn test:app --log-level DEBUG`

Curling the files:
```
$ curl 'http://localhost:8000/static/hello-file'
Hello, World!
$ curl 'http://localhost:8000/static/empty-file'
```
The error is shown for empty-file in Gunicorn logs:
```
[2021-11-22 17:45:20 +0200] [87184] [DEBUG] GET /static/hello-file
[2021-11-22 17:45:23 +0200] [87184] [DEBUG] GET /static/empty-file
[2021-11-22 17:45:23 +0200] [87184] [ERROR] Error handling request
Traceback (most recent call last):
  File "/home/local/jmminkin/tmp/venv/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 182, in handle_request
    resp.write_file(respiter)
  File "/home/local/jmminkin/tmp/venv/lib/python3.8/site-packages/gunicorn/http/wsgi.py", line 385, in write_file
    if not self.sendfile(respiter):
  File "/home/local/jmminkin/tmp/venv/lib/python3.8/site-packages/gunicorn/http/wsgi.py", line 375, in sendfile
    self.sock.sendfile(respiter.filelike, count=nbytes)
  File "/usr/lib/python3.8/socket.py", line 482, in sendfile
    return self._sendfile_use_sendfile(file, offset, count)
  File "/usr/lib/python3.8/socket.py", line 346, in _sendfile_use_sendfile
    self._check_sendfile_params(file, offset, count)
  File "/usr/lib/python3.8/socket.py", line 460, in _check_sendfile_params
    raise ValueError(
ValueError: count must be a positive integer (got 0)
[2021-11-22 17:45:23 +0200] [87184] [DEBUG] Closing connection. 
```

Environment is Python 3.8.10 on Ubuntu 20.04 with the following packages:
```
Package       Version
------------- -------
click         8.0.3
Flask         2.0.2
gunicorn      20.1.0
itsdangerous  2.0.1
Jinja2        3.0.3
MarkupSafe    2.0.1
pip           21.3.1
pkg_resources 0.0.0
setuptools    44.0.0
Werkzeug      2.0.2
```

Running the app on plain Flask does not log the error. In our real usage this error is shown for loading a file for Swagger documentation (`/v1/ui/css/typography.css`).

**Comment by cash** [comment]
This is coming from python's socket code from this function:

```
    def _check_sendfile_params(self, file, offset, count):
        if 'b' not in getattr(file, 'mode', 'b'):
            raise ValueError("file should be opened in binary mode")
        if not self.type & SOCK_STREAM:
            raise ValueError("only SOCK_STREAM type sockets are supported")
        if count is not None:
            if not isinstance(count, int):
                raise TypeError(
                    "count must be a positive integer (got {!r})".format(count))
            if count <= 0:
                raise ValueError(
                    "count must be a positive integer (got {!r})".format(count))
```

**Comment by hazho** [comment]
@cash true, but we can solve it by simply wrapping the command at line: 375 into a conditional statement, I did it like the following:
```

        if nbytes > 0:
            self.sock.sendfile(respiter.filelike, count=nbytes)
```
and voalla, it kicks out the raised error, if you think this is a proper way, then I will make a PR

**Comment by hazho** [comment]
kindly ckeck PR #2773

**Comment by benoitc** [comment]
fixed by #2773. Sorry for the delay to apply it.
